File: dataset/ourMethod/OPENAI_gpt-5-2025-08-07/Task12.java
import java.nio.charset.StandardCharsets;

public final class Task12 {

    private static final int MAX_LEN = 1_000_000;

    public static String highestScoringWord(String s) {
        if (s == null) {
            throw new IllegalArgumentException("Input cannot be null");
        }
        if (s.length() > MAX_LEN) {
            throw new IllegalArgumentException("Input too long");
        }

        int bestStart = -1;
        int bestLen = 0;
        int bestScore = -1;

        int curStart = -1;
        int curLen = 0;
        int curScore = 0;

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (!Character.isWhitespace(ch)) {
                if (curLen == 0) {
                    curStart = i;
                }
                int sc = (ch >= 'a' && ch <= 'z') ? (ch - 'a' + 1) : 0;
                curScore += sc;
                curLen++;
            } else {
                if (curLen > 0) {
                    if (curScore > bestScore) {
                        bestScore = curScore;
                        bestStart = curStart;
                        bestLen = curLen;
                    }
                    curLen = 0;
                    curScore = 0;
                }
            }
        }

        if (curLen > 0) {
            if (curScore > bestScore) {
                bestScore = curScore;
                bestStart = curStart;
                bestLen = curLen;
            }
        }

        if (bestStart == -1) {
            return "";
        }
        return s.substring(bestStart, bestStart + bestLen);
    }

    public static void main(String[] args) {
        String[] tests = new String[] {
            "man i need a taxi up to ubud",
            "what time are we climbing up the volcano",
            "take me to semynak",
            "aa b",
            "abad zzz a b c"
        };

        for (String t : tests) {
            String res = highestScoringWord(t);
            System.out.println("Input: \"" + t + "\" -> Highest: \"" + res + "\"");
        }
    }
}

File: dataset/ourMethod/OPENAI_gpt-5-2025-08-07/Task12.py
from typing import Optional

MAX_LEN = 1_000_000

def highest_scoring_word(s: str) -> str:
    if not isinstance(s, str):
        raise ValueError("Input must be a string")
    if len(s) > MAX_LEN:
        raise ValueError("Input too long")

    best_start = -1
    best_len = 0
    best_score = -1

    cur_start = -1
    cur_len = 0
    cur_score = 0

    for i, ch in enumerate(s):
        if not ch.isspace():
            if cur_len == 0:
                cur_start = i
            sc = (ord('a') <= ord(ch) <= ord('z')) and (ord(ch) - ord('a') + 1) or 0
            cur_score += sc
            cur_len += 1
        else:
            if cur_len > 0:
                if cur_score > best_score:
                    best_score = cur_score
                    best_start = cur_start
                    best_len = cur_len
                cur_len = 0
                cur_score = 0

    if cur_len > 0:
        if cur_score > best_score:
            best_score = cur_score
            best_start = cur_start
            best_len = cur_len

    if best_start == -1:
        return ""
    return s[best_start:best_start + best_len]


if __name__ == "__main__":
    tests = [
        "man i need a taxi up to ubud",
        "what time are we climbing up the volcano",
        "take me to semynak",
        "aa b",
        "abad zzz a b c",
    ]
    for t in tests:
        res = highest_scoring_word(t)
        print(f'Input: "{t}" -> Highest: "{res}"')